sdk/sdk-skeleton.go

```diff
@@ -0,0 +1,126 @@
+import "time"
+
+type Asset struct{}
+
+//// Methods directly on the SDK
+
+func NewChannel() {
+
+}
+
+func NewChannelListenerService() {
+
+}
+
+func GetChannel() {
+
+}
+
+//// Channel State Machine
+
+type ChannelStateMachine struct {
+	ChannelID              string
+	Status                 string // waiting | active | closing | closed
+	Asset                  Asset
+	InitiatorEscrowAccount string
+	ResponderClaimAmount   int
+	ReserveEscrowAccount   string
+	InitiatorClaimAmount   int
+	IsInitiator            bool
+	MyAccount              string
+	MyClaimAMount          int
+	OtherAccount           string
+	OtherClaimAmount       int
+}
+
+func (c *ChannelStateMachine) Init() {
+	return nil
+}
+
+func (c *ChannelStateMachine) RegisterMonitoringNotificationHandler(handler MonitoringNotificationHandler) error {
+	return nil
+}
+
+func (c *ChannelStateMachine) CheckChannel() (*ChannelCheckResponse, error) {
+	return nil, error
+}
+
+func (c *ChannelStateMachine) Payment(sendAmount int, timeoutMillis int) error {
+	return nil
+}
+
+func (c *ChannelStateMachine) CloseStart(declareID int) error {
+	return nil
+}
+
+func (c *ChannelStateMachine) CloseCoordinated(timeoutMillis int, id string) (newStatus string, err error) {
+	return "", nil
+}
+
+func (c *ChannelStateMachine) CloseUncoordinated(id string) error {
+	return nil
+}
+
+func (c *ChannelStateMachine) GetLatestDeclarationTx() (*TxInfo, error) {
+	return nil, nil
+}
+
+func (c *ChannelStateMachine) GetLatestCloseTx(id string) (*TxInfo, error) {
+	return nil, nil
+}
+
+type TxInfo struct {
+	ID               string
+	Type             string // declaration | close
+	Seq              int64
```

Comment:
I think we should include the iteration number here too.

Reply:
https://github.com/stellar/experimental-payment-channels/pull/25/commits/ce73653f4ebdd5884d37aa91946d147fe348610a